compiler/modules/front/resolveTypes.bal

```diff
@@ -254,11 +254,9 @@ function resolveTypeDesc(ModuleSymbols mod, s:ModuleLevelDefn modDefn, int depth
     if td is s:ErrorTypeDesc {
         return t:errorDetail(check resolveTypeDesc(mod, modDefn, depth, td.detail));
     }
-    if td is s:UnaryTypeDesc {
-        t:SemType ty = check resolveTypeDesc(mod, modDefn, depth, td.td);
-        return t:complement(ty);
-    }
-    panic error("unimplemented type-descriptor");
+    s:UnaryTypeDesc _ = td;
```

Comment:
This is not correct Ballerina code. The narrowed type is not s:UnaryTypeDesc (because of mutability)

Reply:
When you narrow with a mutable uniform type, you have to use roDiff not diff. We should try to find a fix that keeps jBallerina happy now, but is nonetheless correct.

Reply:
We can instead do:

```
We can do

```
s:TypeDesc td2 = td;
if td2 is s:UnaryTypeDesc {
```

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/33722